src/views/authenticated-view/index.tsx

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react'
+import { useRouter } from 'next/router'
+import useAuthentication from 'hooks/use-authentication'
+import { AuthenticatedViewProps } from './types'
+
+/**
+ * A view that handles gating content behind authentication. Also handles
+ * redirecting the user to the specified `redirectTo` page if they are not
+ * authenticated.
+ */
+const AuthenticatedView = ({
+	children,
+	redirectTo = '/auth',
+}: AuthenticatedViewProps) => {
+	const router = useRouter()
+	const { isLoading, isAuthenticated, user } = useAuthentication()
+
+	// Redirect to the specified page if not authenticated
+	useEffect(() => {
+		if (!isLoading && !isAuthenticated) {
+			router.push(redirectTo)
+		}
+	}, [isAuthenticated, isLoading, redirectTo, router])
+
+	// Show loading message while data is loading
+	if (!user) {
+		return <p>Loading...</p>
+	}
```

Comment:
Question: Does this handle the SSR case? So initially we see `Loading...` then once everything resolves on the client in theory it would redirect or render its children?

Reply:
Ooohh I'm going to go check out those docs you linked and see what I find!